Tribler/Test/Core/Modules/Api/base_api_test.py

```diff
@@ -0,0 +1,47 @@
+import json
+from twisted.internet.defer import succeed
+from twisted.web.client import Agent, readBody
+from twisted.web.http_headers import Headers
+from Tribler.Core.Utilities.network_utils import get_random_port
+from Tribler.Core.Utilities.twisted_thread import reactor
+from Tribler.Core.version import version_id
+from Tribler.Test.test_as_server import TestAsServer
+
+
+class AbstractApiTest(TestAsServer):
+    """
+    Tests for the Tribler HTTP API should create a subclass of this class.
+    This class contains some auxiliary methods to perform requests and to check the right response code/
+    response json returned.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(AbstractApiTest, self).__init__(*args, **kwargs)
+        self.expected_response_code = 200
+        self.expected_response_json = None
+
+    def setUpPreSession(self):
+        super(AbstractApiTest, self).setUpPreSession()
+        self.config.set_http_api_enabled(True)
+        self.config.set_http_api_port(get_random_port())
+        self.config.set_megacache(True)
+
+    def parse_body(self, body):
+        if body is not None:
+            self.assertDictEqual(json.loads(body), self.expected_response_json)
+
+    def parse_response(self, response):
+        self.assertEqual(response.code, self.expected_response_code)
+        if response.code == 200:
```

Comment:
Not always, see for instance one of my tests (`test_my_channel_overview_endpoint_no_my_channel`)